config/routes.rb

```diff
@@ -2,7 +2,15 @@
   get '/users', to: 'users#index', as: 'users'
   get '/users/:id', to: 'users#show', as: 'user'
   get '/users/:user_id/posts', to: 'posts#index', as: 'user_posts'
+  get '/users/:user_id/posts/new', to: 'posts#new', as: 'new_post'
+  post '/users/:user_id/posts', to: 'posts#create', as: 'create_post'
   get '/users/:user_id/posts/:id', to: 'posts#show', as: 'user_post'
+  resources :users, only: [] do
+  resources :posts, only: [] do
+    resources :comments, only: [:new, :create]
+    resources :likes, only: [:new, :create]
+  end
+end
```

Comment:
- [ ] [optional] @cosywasswa It would have been nice if you considered adding a root route in your routes and also refactor your code to only use resources this will make your code easier to read and scale in future
```suggestion
  resources :users, only: [;index, :show, ] do
  resources :posts, only: [:index, :new, :create, :show] do
    resources :comments, only: [:new, :create]
    resources :likes, only: [:new, :create]
  end
end
```